burner-wallet-plugin/tokenbridge-bw-exchange/src/utils/utils.ts

```diff
@@ -34,11 +34,23 @@ export const waitForEvent = async (web3, contract: Contract, event: string, call
   }
 }
 
-export const isBridgeContract = async (contract: Contract): Promise<boolean> => {
+export const isVanillaBridgeContract = async (contract: Contract): Promise<boolean> => {
   try {
     await contract.methods.deployedAtBlock().call()
     return true
   } catch (e) {
     return false
   }
 }
+
+export const isBridgeContract = async (contract: Contract, allowedModes?: string[]): Promise<boolean> => {
+  try {
+    const mode = await contract.methods.getBridgeMode().call()
```

Comment:
The method will return 4 bytes at the same time `allowedModes.include` will compare this value with a set of strings. Is my understanding correct? How it should work in this case?

Reply:
When parsing a call returned data, web3js treats `bytes4` as a string (i. e. `0xe177c00f`), so it is ok to compare it with other strings.

Reply:
Sorry, but I don't understand how `0xe177c00f` will match with `erc-to-native-amd` for example.

Reply:
For that, we have a custom getter and setter behind `bridgeModes` field: https://github.com/poanetwork/tokenbridge/blob/3d3374249dfe4beebbaf9a3b0c1661da654a8f8a/burner-wallet-plugin/tokenbridge-bw-exchange/src/burner-wallet/assets/BridgeableERC20Asset.ts#L16-L22

Reply:
Seems I have missed it